components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt

```diff
@@ -18,13 +19,24 @@ class BrowserToolbar @JvmOverloads constructor(
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0
 ) : FrameLayout(context, attrs, defStyleAttr), Toolbar {
-    private val urlView = TextView(context)
 
-    init {
-        addView(urlView)
+    override fun displayUrl(url: String) {
+        val urlView = getUrlViewComponent()
+        urlView.setText(url)
+        urlView.setSelection(0, url.length)
     }
 
-    override fun displayUrl(url: String) {
-        urlView.text = url
+    override fun setOnUrlChangeListener(listener: (String) -> Unit) {
+        val urlView = getUrlViewComponent()
+        urlView.setOnEditorActionListener { _, actionId, _ ->
+            if (actionId == EditorInfo.IME_ACTION_GO) {
+                listener.invoke(urlView.text.toString())
+            }
+            true
+        }
+    }
+
+    internal fun getUrlViewComponent(): EditText {
+        return this.findViewById(R.id.urlView)
     }
```

Comment:
Not important for now... But I think the layout (and structure) for the toolbar should live in this module and not in the app.

Reply:
Agreed, BrowserToolBar should come with a functioning structure ;). I was mainly looking to make sure consumers can specify their own urlView if they wanted to.